src/hotspot/cpu/x86/gc/shared/barrierSetAssembler_x86.hpp

```diff
@@ -50,6 +50,8 @@ class BarrierSetAssembler: public CHeapObj<mtGC> {
 
   virtual void value_copy(MacroAssembler* masm, DecoratorSet decorators,
                           Register src, Register dst, Register value_klass);
+  virtual void flat_field_copy(MacroAssembler* masm, DecoratorSet decorators,
+                          Register src, Register dst, Register inline_layout_info);
```

Comment:
Does new flat_field_copy eventually replace value_copy ? Yes, this checks out (re-read the PR comments)